src/poetry/utils/env/env_manager.py

```diff
@@ -405,10 +405,34 @@ def remove(self, python: str) -> Env:
                     del envs[base_env_name]
                     envs_file.write(envs)
 
-        self.remove_venv(venv_path)
+        self.remove_venv_dir(venv_path)
 
         return VirtualEnv(venv_path, venv_path)
 
+    def remove_venv(self, venv: VirtualEnv) -> Env:
+        venv_path = self._poetry.config.virtualenvs_path
+        cwd = self._poetry.file.path.parent
+        envs_file = TOMLFile(venv_path / self.ENVS_FILE)
+        base_env_name = self.generate_env_name(self._poetry.package.name, str(cwd))
+
+        env_path = venv.path
+        _, minor, _ = venv.get_version_info()
+
+        name = f"{base_env_name}-py{minor}"
+
+        if not env_path.exists():
+            raise ValueError(f'<warning>Environment "{name}" does not exist.</warning>')
+
+        if envs_file.exists():
+            envs = envs_file.read()
+            if base_env_name in envs:
+                del envs[base_env_name]
+                envs_file.write(envs)
+
+        self.remove_venv_dir(env_path)
+
+        return VirtualEnv(env_path, env_path)
```

Comment:
Why do we return a venv?

Reply:
Again, I sticked to what appears to be the norm in the Poetry repository. The original `.remove()` method returns a virtual env, thus I kept that design parttern here too.